test/specs/integration.spec.js

```diff
@@ -65,125 +62,116 @@ describe('Integration tests', function() {
+        done();
+      }, 500);
     });
-    waitsFor(function() { return cm2.getValue() === '3456789' }, 'Got correct end value.');
   });
 
-  it('Random text changes', function () {
+  it('Random text changes', function (done) {
     var ref = new Firebase('https://firepad-test.firebaseio-demo.com').push();
-    var cm1 = CodeMirror(cmDiv());
-    var cm2 = CodeMirror(cmDiv());
+    var cm1 = CodeMirror(hiddenDiv());
+    var cm2 = CodeMirror(hiddenDiv());
     var firepad1 = new Firepad(ref, cm1);
     var firepad2 = new Firepad(ref, cm2);
 
-    var n = 25;
-
-    var done = false;
-    function step () {
-      if (n == 0) {
-        done = true;
-        return;
+    function step (times) {
+      if (times == 0) {
+        done();
+      }else{
+        randomOperation(cm1);
+        setTimeout(function() {
+          expect(cm1.getValue()).toEqual(cm2.getValue());
+          step(times - 1);
+        }, 400);
       }
-
-      n--;
-      randomOperation(cm1);
-
-      waitsFor(function() { return cm1.getValue() == cm2.getValue(); }, 'firepads to match');
-      runs(step);
     }
 
-    var ready = false;
     firepad1.on('ready', function() {
       firepad1.setText('lorem ipsum');
-      ready = true;
+      step(25);
     });
+  }, 777 * 25 + 1000);
 
-    waitsFor(function() { return ready; }, 'Firepad ready.');
-
-    runs(step);
-  });
-
-  it('Performs getHtml responsively', function() {
+  it('Performs getHtml responsively', function(done) {
     var ref = new Firebase('https://firepad-test.firebaseio-demo.com').push();
-    var cm = CodeMirror(cmDiv());
+    var cm = CodeMirror(hiddenDiv());
     var firepad = new Firepad(ref, cm);
 
-    waitsFor(function() { return firepad.ready_ }, 'firepad is ready');
-
-    var html = '<b>bold</b>';
-    runs(function() {
+    firepad.on('ready', function() {
+      var html = '<b>bold</b>';
       firepad.setHtml(html);
       expect(firepad.getHtml()).toContain(html);
+      done();
     });
   });
 
-  it('Uses defaultText to initialize the pad properly', function() {
+  it('Uses defaultText to initialize the pad properly', function(done) {
     var ref = new Firebase('https://firepad-test.firebaseio-demo.com').push();
-    var cm = CodeMirror(cmDiv());
-    var cm2 = CodeMirror(cmDiv());
+    var cm = CodeMirror(hiddenDiv());
+    var cm2 = CodeMirror(hiddenDiv());
     var text = 'This should be the starting text';
     var text2 = 'this is a new, different text';
     var firepad = new Firepad(ref, cm, { defaultText: text});
-    var firepad2 = null;
 
-    waitsFor(function() { return firepad.ready_ }, 'firepad is ready');
-
-    runs(function() {
+    firepad.on('ready', function() {
       expect(firepad.getText()).toEqual(text);
       firepad.setText(text2);
-      firepad2 = new Firepad(ref, cm2, { defaultText: text});
+      var firepad2 = new Firepad(ref, cm2, { defaultText: text});
+      firepad2.on('ready', function() {
+        if(firepad2.getText() == text2){
```

Comment:
Can you add spaces to match the rest of the formatting in this file?  E.g.:

    if (firepad2.getText() == text2) {